internal/credit/creditdriver/balance.go

```diff
@@ -61,3 +66,73 @@ func (b *builder) GetLedgerBalance() GetLedgerBalanceHandler {
 			}))...,
 	)
 }
+
+func mapBalanceToAPI(balance credit.Balance, highwatermark credit.HighWatermark) api.LedgerBalance {
+	var featureBalances []api.FeatureBalance = make([]api.FeatureBalance, 0, len(balance.FeatureBalances))
+	for _, featureBalance := range balance.FeatureBalances {
+		featureBalances = append(featureBalances, mapFeatureBalanceToAPI(featureBalance))
+	}
+
+	var grantBalances []api.LedgerGrantBalance = make([]api.LedgerGrantBalance, 0, len(balance.GrantBalances))
+	for _, grantBalance := range balance.GrantBalances {
+		grantBalances = append(grantBalances, mapGrantBalanceToAPI(grantBalance))
+	}
+
+	return api.LedgerBalance{
+		FeatureBalances: featureBalances,
+		GrantBalances:   grantBalances,
+		LastReset:       &highwatermark.Time,
+		Metadata:        &balance.Metadata,
+		Subject:         balance.Subject,
+	}
+}
+
+func mapFeatureBalanceToAPI(featureBalance credit.FeatureBalance) api.FeatureBalance {
+	id := new(string)
+	if featureBalance.ID != nil {
+		*id = string(*featureBalance.ID)
+	}
+	return api.FeatureBalance{
+		Archived:            featureBalance.Archived,
+		CreatedAt:           featureBalance.CreatedAt,
+		Id:                  id,
+		MeterGroupByFilters: featureBalance.MeterGroupByFilters,
+		MeterSlug:           featureBalance.MeterSlug,
+		Name:                featureBalance.Name,
+		UpdatedAt:           featureBalance.UpdatedAt,
+		Usage:               featureBalance.Usage,
+		Balance:             featureBalance.Balance,
+	}
+}
+
+func mapGrantBalanceToAPI(grantBalance credit.GrantBalance) api.LedgerGrantBalance {
+	var id *string
+	if grantBalance.ID != nil {
+		*id = string(*grantBalance.ID)
+	}
+	var parentId *string
+	if grantBalance.ParentID != nil {
+		*parentId = string(*grantBalance.ParentID)
```

Comment:
I think this will cause panic, as parentId is nil

Reply:
it wont https://go.dev/play/p/qa8fpV1Zffp